homeassistant/components/elkm1/__init__.py

```diff
@@ -117,27 +122,47 @@ def _included(ranges, set_to, values):
                 raise vol.Invalid("Invalid range {}".format(rng))
             values[rng[0]-1:rng[1]] = [set_to] * (rng[1] - rng[0] + 1)
 
-    conf = hass_config[DOMAIN]
-    config = {'temperature_unit': conf[CONF_TEMPERATURE_UNIT]}
-    config['panel'] = {'enabled': True, 'included': [True]}
-
-    for item, max_ in configs.items():
-        config[item] = {'enabled': conf[item][CONF_ENABLED],
-                        'included': [not conf[item]['include']] * max_}
-        try:
-            _included(conf[item]['include'], True, config[item]['included'])
-            _included(conf[item]['exclude'], False, config[item]['included'])
-        except (ValueError, vol.Invalid) as err:
-            _LOGGER.error("Config item: %s; %s", item, err)
-            return False
-
-    elk = elkm1.Elk({'url': conf[CONF_HOST], 'userid': conf[CONF_USERNAME],
-                     'password': conf[CONF_PASSWORD]})
-    elk.connect()
-
-    _create_elk_services(hass, elk)
-
-    hass.data[DOMAIN] = {'elk': elk, 'config': config, 'keypads': {}}
+    for index, conf in enumerate(hass_config[DOMAIN]):
+        _LOGGER.debug("Setting up elkm1 #%d - %s", index, conf['host'])
+
+        config = {'temperature_unit': conf[CONF_TEMPERATURE_UNIT]}
+        config['panel'] = {'enabled': True, 'included': [True]}
+
+        for item, max_ in configs.items():
+            config[item] = {'enabled': conf[item][CONF_ENABLED],
+                            'included': [not conf[item]['include']] * max_}
+            try:
+                _included(conf[item]['include'], True,
+                          config[item]['included'])
+                _included(conf[item]['exclude'], False,
+                          config[item]['included'])
+            except (ValueError, vol.Invalid) as err:
+                _LOGGER.error("Config item: %s; %s", item, err)
+                return False
+
+        prefix = conf.get(CONF_PREFIX, "")
+        device = devices.get(prefix)
+        if device is not None:
+            if prefix == "":
+                _LOGGER.error("At most one elk m1 configuration may skip " +
```

Comment:
We should write a custom config schema validator that checks that prefix is unique and set if more than one device is configured.

See the mysensors component for example.